torch/csrc/jit/codegen/cuda/codegen.cpp

```diff
@@ -581,6 +581,14 @@ class CudaKernelGenerator : private kir::IrVisitor {
 
   void visit(const kir::BroadcastOp* node) final {
     TORCH_INTERNAL_ASSERT(node->out()->isA<kir::TensorIndex>());
+
+    // Emit a SET operation if the broadcast has been fused
+    //  into the warp reduction producer
+    if (node->isOmitted()) {
```

Comment:
Can we just remove omitted broadcast expressions during lowering?

Reply:
Great question. I actually tried and it wasn't straightforward to just remove from IR. The main difficulty comes from reduction ops with broadcast axes that'd result from removing.

Reply:
Do you remember what the problem looked like?

Reply:
Yes. There might be a way that could have worked but I don't know if there's a clean enough way. What I tried was removing the broadcast on fusion IR, and then 
```cpp
  T1[I R R] = reduce(T0[I I I])
  T2[I B B] = broadcast(T1[I R R])
```

becomes one of the following:
```
 T2[I R R] = reduce(T0[I I I])   // This is incorrect on the consumer side
```
```
 T2[I B B] = reduce(T0[I I I])   // This is incorrect on the producer side
```
```
 T2[I B B R R] = reduce(T0[I I I])   // This is could have worked mechanically but doesn't really make too much sense
```

I guess trying to remove them on kernel IR after indexing could also have been another option but I think converting to a `SET` is simpler.

Reply:
I think doing so in the Fusion IR would be difficult. It would invalidate some of the assumptions we make when analyzing the IR. What I was thinking was to do some clean-up pass at the end of lowering. A that point, I believe we could remove the broadcast expression and replace all the uses of the broadcast output by the reduce output. That said, it doesn't seem to make much of a difference, so I'm fine with the API. Perhaps, `isFused` may better capture what's actually done?

Reply:
I see. I agree that removing on kernel IR at a later stage could work too. I guess I could try that later if we need. 

I actually thought about exactly `isFused`. I will rename it that way. 

Thanks for the suggestions.

Reply:
I think this approach is fine for now, though I think cleanest approach would be have a simple pass to convert the operation in the kernel ir.